sledgehammer/start-up.sh

```diff
@@ -107,28 +114,43 @@ else
     fi
 fi
 
+IPv6=0
 if ! [[ $(ip -4 -o addr show dev $BOOTDEV) =~ $ip_re ]]; then
-    echo "We did not get an address on $BOOTDEV"
-    echo "Things will end badly."
+    if [[ $(ip -6 -o addr show dev $BOOTDEV) =~ $ip6_re ]]; then
+        IPv6=1
+    else
+        echo "We did not get an address on $BOOTDEV"
+        echo "Things will end badly."
+    fi
 fi
 MYIP="${BASH_REMATCH[1]}"
 
 if suse_ver 12; then
     [ -f /etc/sysconfig/network/config ] && source /etc/sysconfig/network/config
     WAIT_FOR_INTERFACES=${WAIT_FOR_INTERFACES:-120}
     /usr/lib/wicked/bin/wickedd-dhcp4 --test --test-output /tmp/wicked-dhcp-$BOOTDEV --test-timeout $WAIT_FOR_INTERFACES $BOOTDEV
+    if (( $? > 0 )); then
```

Comment:
Anti-pattern.

if ! /usr/lib/wicked/bin/wickedd-dhcp4 --test --test-output /tmp/wicked-dhcp-$BOOTDEV --test-timeout $WAIT_FOR_INTERFACES $BOOTDEV; then

*However* since the command-line arguments so incredibly similar, refactor them into DHCP_ARGUMENTS or such

Reply:
done